src/reset-delay.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+
+const useResetDelay = ({
+	resetDelay,
+	setData,
+	setError,
+	setIsFetched,
+	setIsFetching,
+	timerSignal
+}) =>
+	useEffect(() => {
+		let timer;
+		if (timerSignal && resetDelay) {
+			timer = setTimeout(() => {
+				// reset the state
+				setData(null);
+				setIsFetching(false);
+				setIsFetched(false);
+				setError(null);
+			}, resetDelay);
+		}
+
+		return () => timer && clearTimeout(timer);
+	}, [resetDelay, setData, setError, setIsFetched, setIsFetching, timerSignal]);
```

Comment:
This used to just have `resetDelay` & `timerSignal` as deps because he other functions were coming from the state hook. It seems the linter can't recognize that now, but does it affect anything at runtime? I would think it would be ok...

Reply:
Tests still pass!